pkg/controller/volume/attachdetach/attach_detach_controller.go

```diff
@@ -649,6 +667,65 @@ func (adc *attachDetachController) processVolumesInUse(
 	}
 }
 
+// installCRDs creates the specified CustomResourceDefinition for the CSIDrivers object.
+func (adc *attachDetachController) installCRDs() error {
+	crd := &apiextensionsv1beta1.CustomResourceDefinition{
+		ObjectMeta: metav1.ObjectMeta{
+			Name: csiapiv1alpha1.CsiDriverResourcePlural + "." + csiapiv1alpha1.GroupName,
+		},
+		Spec: apiextensionsv1beta1.CustomResourceDefinitionSpec{
+			Group:   csiapiv1alpha1.GroupName,
+			Version: csiapiv1alpha1.SchemeGroupVersion.Version,
+			Scope:   apiextensionsv1beta1.ClusterScoped,
+			Names: apiextensionsv1beta1.CustomResourceDefinitionNames{
+				Plural: csiapiv1alpha1.CsiDriverResourcePlural,
+				Kind:   reflect.TypeOf(csiapiv1alpha1.CSIDriver{}).Name(),
+			},
+		},
```

Comment:
Why not use the `validation` property? I think you should. I think there is a way to construct it from the types.go file you have.

Reply:
Oh nice, I didn't realize validation was already supported. Looks like auto generation of OpenAPIV3Schema  from types.go is still in the works https://github.com/kubernetes/kube-openapi/pull/37. Before we move to beta, hopefully that is ready and we can use it. If not, we'll just construct the Validation schema manually then.